In [6]:
import pandas as pd
import numpy as np
df = pd.read_csv(r"E:\5 Python\property.csv") 
df.head()

Suburb           Address  Rooms Type      Price Method SellerG  \
0  Abbotsford      85 Turner St      2    h  1480000.0      S  Biggin   
1  Abbotsford   25 Bloomburg St      2    h  1035000.0      S  Biggin   
2  Abbotsford      5 Charles St      3    h  1465000.0     SP  Biggin   
3  Abbotsford  40 Federation La      3    h   850000.0     PI  Biggin   
4  Abbotsford       55a Park St      4    h  1600000.0     VB  Nelson   

        Date  Distance  Postcode  ...  Bathroom  Car  Landsize  BuildingArea  \
0  3/12/2016       2.5    3067.0  ...       1.0  1.0     202.0           NaN   
1  4/02/2016       2.5    3067.0  ...       1.0  0.0     156.0          79.0   
2  4/03/2017       2.5    3067.0  ...       2.0  0.0     134.0         150.0   
3  4/03/2017       2.5    3067.0  ...       2.0  1.0      94.0           NaN   
4  4/06/2016       2.5    3067.0  ...       1.0  2.0     120.0         142.0   

   YearBuilt  CouncilArea Lattitude  Longtitude             Regionname  \
0        NaN        Yarra  -37.7996    144.9984  Northern Metropolitan   
1     1900.0        Yarra  -37.8079    144.9934  Northern Metropolitan   
2     1900.0        Yarra  -37.8093    144.9944  Northern Metropolitan   
3        NaN        Yarra  -37.7969    144.9969  Northern Metropolitan   
4     2014.0        Yarra  -37.8072    144.9941  Northern Metropolitan   

  Propertycount  
0        4019.0  
1        4019.0  
2        4019.0  
3        4019.0  
4        4019.0  

[5 rows x 21 columns]

## Q3 : For the suburb of Abbotsford, what is the probability that out of 10 properties sold, 3 will not have car parking? Use the column car in the dataset. Round off your answer to 3 decimal places.

In [2]:
df['Car'].unique()

array([ 1.,  0.,  2.,  6.,  5.,  4.,  3.,  8.,  7.,  9., 10., nan])

In [3]:
# Car: Impute missing values with the mode_Car
mode_Car = df['Car'].mode()[0]
df['Car'].fillna(mode_Car, inplace=True)
df['Car'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 13580 entries, 0 to 13579
Series name: Car
Non-Null Count  Dtype  
--------------  -----  
13580 non-null  float64
dtypes: float64(1)
memory usage: 106.2 KB


In [4]:
df['Car'].unique()

array([ 1.,  0.,  2.,  6.,  5.,  4.,  3.,  8.,  7.,  9., 10.])

In [5]:
# Filter data for Abbotsford
abbotsford_data = df[df['Suburb'] == 'Abbotsford']

# Calculate p (probability of no car parking)
total_properties = len(abbotsford_data)
no_car_parking = len(abbotsford_data[abbotsford_data['Car'] == 0])
p = no_car_parking / total_properties
n = 10 
k = 3
probability = stats.binom.pmf(k,n,p)
print(f"Total properties in Abbotsford: {total_properties}")
print(f"Properties with No Car Parking: {no_car_parking}")
print(f"Probability of exactly 3 out of 10 properties having no car parking: {probability : .3f}")

Total properties in Abbotsford: 56
Properties with No Car Parking: 15
Probability of exactly 3 out of 10 properties having no car parking:  0.260


- I used the binomial probability distribution to calculate the probability of selecting exactly 3 properties without car parking from a sample of 10 properties.
- n=10 (number of trials),
- k=3 (number of successes, i.e., properties with no car parking),
- p=0.2679 (probability of selecting a property with no car parking),
- 1−p=0.7321 (probability of selecting a property with car parking).

**Conclusion:**
- The probability of exactly 3 properties out of 10 having no car parking is relatively moderate at 26%. This means that there is a 26% chance that exactly 3 out of 10 properties selected will have no car parking. This suggests that while properties without car parking are not the majority, they are still fairly common in Abbotsford. This information could be useful for real estate professionals, potential buyers, or investors considering the availability and desirability of properties with car parking in the area.